.github/workflows/release.yml

```diff
@@ -109,20 +110,43 @@ jobs:
         with:
           version: ${{ inputs.version }}
           tag_message_template: 'Release ${VERSION}'
+          # Don't push tag, we'll do that after merging up
+          push_tag: false
 
       - name: "Bump to next development release and commit"
         uses: mongodb-labs/drivers-github-tools/bump-version@v2
         with:
           version: ${{ inputs.version }}
           version_bump_script: "./bin/update-release-version.php to-next-patch-dev"
           commit_template: 'Back to -dev'
+          # Don't push commit, we still need to merge up
+          push_commit: false
+
+      - name: "Determine branch to merge up to"
+        id: get-next-branch
+        uses: alcaeus/automatic-merge-up-action/get-next-branch@main
```

Comment:
I'll keep the action separate for the time being. Once I've confirmed this works I'll tag the current state at v1 and update all of our repositories accordingly.

As for the 20 KLOC, all of the dependencies are compiled into a single file so that running the action from a workflow doesn't require installing them manually. The main dependencies are the actions toolkit that makes dealing with inputs, outputs, and the GitHub API easier.